woocommerce/woocommerce-ios
# Order: Enable copying details on billing info screen

Closes: #11369 


## What

This PR updates the billing info screen to simplify copying address, phone number and email address.

## Why
Merchants have been complaining about the lack of support for contacting their customers through different messaging apps. This is not technically possible on iOS due to privacy restrictions, but we can make the experience easier by letting them copy billing details.

The current behavior is not great:
- The address field requires long tap gesture to display the copy option.
- The phone field doesn't have the copy option.
- The email field shows the copy option only when the Mail app is not available.

## How
- Added an action for the phone row to copy the number.
- Update the action handler for the email row to show an action sheet with an option to copy the email address instead of enabling this only when the Mail app is unavailable.
- Updated the copy option on the address field to make it available upon a single tap on the row.
- Display a notice when a detail has been copied to the clipboard for clarity.

## Testing instructions

- Log in to a store and navigate to the Orders tab.
- Select an existing order with all details: name, billing address, phone, email or create a new one.
- On the order detail screen, select Billing information.
- Tap the address row, notice that the address is copied to clipboard instantly. Confirm that an event is tracked: `order_detail_customer_address_tapped `.
- Tap the phone row, notice that there is an option to copy the number.
- Tap the Copy number option, notice that an event is tracked: `order_detail_customer_info_phone_menu_copy_tapped`.
- Tap the email row, notice that there is an action sheet displayed with an option to copy the email address. If you are testing on the simulator, the email option will not be available.
- Tap the Copy email address option, notice that an event is tracked: `order_detail_customer_info_email_copy_option_tapped`.
- If the email option is available, tap it and confirm that an event is tracked: `order_detail_customer_info_email_option_tapped`.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/2b582bd2-517a-42dd-a8fc-7e3c79168f22



---
